nixos/modules/services/logging/journaldriver.nix

```diff
@@ -0,0 +1,112 @@
+# This module implements a systemd service for running journaldriver,
+# a log forwarding agent that sends logs from journald to Stackdriver
+# Logging.
+#
+# It can be enabled without extra configuration when running on GCP.
+# On machines hosted elsewhere, the other configuration options need
+# to be set.
+#
+# For further information please consult the documentation in the
+# upstream repository at: https://github.com/aprilabank/journaldriver/
+
+{ config, lib, pkgs, ...}:
+
+with lib; let cfg = config.services.journaldriver;
+in {
+  options.services.journaldriver = {
+    enable = mkOption {
+      type        = types.bool;
+      default     = false;
+      description = ''
+        Whether to enable journaldriver to forward journald logs to
+        Stackdriver Logging.
+      '';
+    };
+
+    logLevel = mkOption {
+      type        = types.string;
```

Comment:
Almost missed this: The type should be `types.str` instead: https://github.com/NixOS/nixpkgs/blob/7296c261f782ae525bdaf2a8c885e0d2c0f640e1/lib/types.nix#L203-L205

The same for all other options that use `types.string` except `applicationCredentials`, which should be changed to a path: `types.path`.

Reply:
Fixed!